### ISSUE TITLE
Screen Readers: Create Expenses: The amount field does not announce the currency

### ISSUE THREAD

**Issue by elisenwazuoke** [Expensify/App#77369]
## Action Performed:
Precondition: The User should be logged in

1. Using Android TackBack or iOS VoiceOver, open the Expensify app
2. Swipe to the Create '+' button and double-tap to activate
3. Swipe to the 'Create Expense' button and double-tap to activate
4. Swipe to the 'Amount' input field and type the amount
5. Listen to the VoiceOver announcement

## Expected Result:
The field should be announced with the displayed currency.

## Actual Result:
The amount filed does not announce the currency.

## Area issue was found in:
Create Expenses

## Failed WCAG checkpoints
WCAG 1.3.1 Info and Relationships
https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships

## User impact:
Users with different disabilities employ assistive technology to adapt content according to their specific needs. For example, users who are blind (using a screen reader) benefit when information conveyed through color is also available in text (including text alternatives for images that use color to convey information), and users who are deaf-blind and use braille (text) refreshable displays may be unable to access color-dependent information.

## Suggested resolution:
Explicitly include the currency unit (e.g., "dollars," "euros") in the field's accessibilityLabel.
https://reactnative.dev/docs/accessibility#accessibilitylabel

## Environment(s)
Samsung Galaxy S24 Ultra - Android 14, Partner - Native (No mobile browser)

## Screenshots/Videos
![Image](https://github.com/user-attachments/assets/a3bdeaae-b538-41b0-8312-074077dd3edf)

https://github.com/user-attachments/assets/e30b77d5-ba0d-4683-a861-66e1fd00d0f4

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7006134
**Issue reported by:** NA
**Slack conversation:** NA
